extension/source/Home/Wallet/Wallets/SendDetail/SendDetail.tsx

```diff
@@ -0,0 +1,141 @@
+    }),
+    [],
+  );
+
+  const $selectedAsset = useCell(cells.selectedAsset);
+  const $recipient = useCell(cells.recipient);
+  const $amountWei = useCell(cells.amountWei);
+
+  const [sendState, setSendState] = useState<SendState>();
+
+  return (
+    <div className="flex flex-col gap-8">
+      <BigSendButton
+        selectedAsset={cells.selectedAsset}
+        recipient={cells.recipient}
+        amountWei={cells.amountWei}
+      />
+      {(() => {
+        if ($selectedAsset === undefined) {
+          return <AssetSelector selectedAsset={cells.selectedAsset} />;
+        }
+
+        if ($recipient === undefined) {
+          return <RecipientSelector recipient={cells.recipient} />;
+        }
+
+        if ($amountWei === undefined) {
+          return (
+            <AmountSelector
+              selectedAsset={cells.selectedAsset}
+              onSend={async (wei) => {
+                cells.amountWei.write(wei);
+
+                const from = await quill.cells.selectedAddress.read();
+                assert(from !== undefined);
+
+                assert($recipient !== undefined);
+
+                try {
+                  const sendBlock = await quill.cells.blockNumber.read();
+                  setSendState({ step: 'sending', sendBlock });
+
+                  let receipt: Receipt | undefined;
+
+                  const txHash = await quill.rpc.eth_sendTransaction({
+                    from,
+                    to: $recipient,
+                    gas: undefined,
+                    gasPrice: undefined,
+                    value: wei,
+                    data: undefined,
+                  });
+
+                  setSendState({
+                    step: 'awaiting-confirmation',
+                    txHash,
+                    sendBlock,
+                  });
+
+                  const forEachHandle = forEach(
+                    quill.cells.blockNumber,
+                    async ($blockNumber) => {
+                      if ($blockNumber > sendBlock) {
+                        receipt = await quill.rpc.eth_getTransactionReceipt(
+                          txHash,
+                        );
+
+                        if (receipt === undefined) {
+                          return;
+                        }
+
+                        forEachHandle.stop();
+                      }
+                    },
+                  );
+
+                  await forEachHandle.iterationCompletionPromise;
+                  assert(receipt !== undefined);
+
+                  setSendState({
+                    step: 'confirmed',
+                    sendBlock,
+                    receipt,
+                  });
+                } catch (error) {
+                  assert(error instanceof Error);
+                  setSendState({ step: 'error', message: error.message });
+                }
+              }}
+            />
+          );
+        }
+
+        return sendState && <SendProgress state={sendState} />;
+      })()}
```

Comment:
Instead of doing this as an inline anonymous component, could be split up into two more components:
 - `SendDetailSelector`, which would just be the early returns of components based on conditions
 - `SendDetailAmountSelector`, which would be returned by `SendDetailSelector` and would contain the `onSend` eth send logic.
You could then pass the top level `cells` down to those children components to keep state in sync.

I had a difficult time initially reading through this component, which was an indication it might be good to break up.

Reply:
Will fix 👍.

Reply:
Fixed